#638 Fix Application Home start message

CarlosNZ opened the pull request:
Fixes #638.

An unevaluated "startMessage" was being passed to ApplicationHome. Have fixed so that both startMessage and submissionMessage are evaluated in `useLoadApplication` and stored in structure.info (previously it was just submissionMessage), so evaluated startMessage string can be available in ApplicationHome.

To test: start a new application. You should see a correctly rendered "Start Message" box on both the "ApplicationCreate" page `/new?...` and the "ApplicationHome" page (`/application/<serial>`).

andreievg reviewed (commented):
Looks good,

Although I am not sure why we have `template` as output of `useLoadApplication`, do you think it's worth removing it in this PR, structure.info.template is available for `ApplicationContainer`, `ApplicationHome` in ApplicationWrapper (that uses useLoadApplication), do you think it' worth removing in this PR ?

CarlosNZ commented:
> Although I am not sure why we have `template` as output of `useLoadApplication`

Yes, I was wondering that too, but I was wary of touching it as I assumed it would be used elsewhere. But it does seem like everything in "template" is also being saved in "structure.info", so perhaps we should remove it?

I'll spend a few minutes trying to make sense of it, and see what the consequences of removing it are. But if it causes other problems, then it's probably best to leave it to another issue, right?

CarlosNZ commented:
> I'll spend a few minutes trying to make sense of it, and see what the consequences of removing it are. But if it causes other problems, then it's probably best to leave it to another issue, right?

@andreievg Okay, I've removed `template` from the output of useLoadApplication, and replaced usage of it with `structure.info.template`, which contains the same info.

Seems to working okay with no noticeable breakages 🤞 c972cb8f

andreievg reviewed (commented):
thanks, one more little change please

useLoadApplication.ts

- 38: const [template, setTemplate] = useState<TemplateDetails>()

- 91 through to 96

CarlosNZ commented:
> thanks, one more little change please
> 
> useLoadApplication.ts
> 
>     * 38: const [template, setTemplate] = useState()
> 
>     * 91 through to 96

Sweet 05a66d3e
